src/main/java/org/elasticsearch/index/analysis/TurkishStemmer.java

```diff
@@ -158,7 +158,7 @@ public TurkishStemmer(final CharArraySet protectedWords,
    * @param len the length of the word
    * @return the stemmed word
    */
-  public String stem(char s[], int len) {
+  public String stem(char s[], int len, int d) {
```

Comment:
Best abstain from changing a public methods API on a minor revision.
Preferably have this be a new method used by another one, that keeps tha same API.

Reply:
We're overloading `stem` method in case someone is using it separately and won't upgrade to the new signature.